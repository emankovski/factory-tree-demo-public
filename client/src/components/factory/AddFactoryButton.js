import React, { useState, Fragment } from 'react'
import PropTypes from 'prop-types'
import Fab from '@material/react-fab'
import MaterialIcon from '@material/react-material-icon'

import FactoryEditor from './FactoryEditor'

export const AddFactoryButton = ({ addFactory }) => {
  const [isEditing, setIsEditing] = useState(false)

  const editorView = isEditing ? (
    <FactoryEditor
      isAddingNew={true}
      cancel={() => setIsEditing(false)}
      save={value => {
        setIsEditing(false)
        addFactory(value)
      }}
    />
  ) : null

  return (
    <Fragment>
      <Fab
        style={{ position: 'fixed', bottom: 20, right: 20, zIndex: 1 }}
        icon={<MaterialIcon icon="add" />}
        onClick={() => setIsEditing(true)}
      ></Fab>
      {editorView}
    </Fragment>
  )
}

AddFactoryButton.propTypes = {
  addFactory: PropTypes.func.isRequired
}

export default AddFactoryButton
