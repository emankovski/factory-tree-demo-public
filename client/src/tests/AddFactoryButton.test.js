import AddFactoryButton from '../components/factory/AddFactoryButton'

import React from 'react'
import { create, act } from 'react-test-renderer'

describe('Add Factory Button', () => {
  test('it renders as button', () => {
    let component
    act(() => {
      component = create(
        <AddFactoryButton
          addFactory={factory => {
            console.log(factory)
          }}
        />
      )
    })
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('handles click', () => {
    let component
    act(() => {
      component = create(
        <AddFactoryButton
          addFactory={factory => {
            console.log(factory)
          }}
        />
      )
    })
    let tree = component.toJSON()
    act(() => {
      tree.props.onClick()
    })
    // re-rendering
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
