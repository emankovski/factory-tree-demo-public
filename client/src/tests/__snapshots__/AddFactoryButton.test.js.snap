// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Factory Button handles click 1`] = `
Array [
  <button
    className="mdc-fab"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "bottom": 20,
        "position": "fixed",
        "right": 20,
        "zIndex": 1,
      }
    }
  >
    <i
      className="material-icons mdc-fab__icon"
    >
      add
    </i>
  </button>,
  <div>
    <div
      aria-describedby="mdc-dialog-content"
      aria-labelledby="mdc-dialog-title"
      aria-modal={true}
      className="mdc-dialog mdc-dialog--opening"
      id="mdc-dialog"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="alertdialog"
    >
      <div
        className="mdc-dialog__container"
      >
        <div
          className="mdc-dialog__surface"
        >
          <h2
            className="mdc-dialog__title"
            id="mdc-dialog-title"
          >
            Add Factory
          </h2>
          <div
            className="mdc-dialog__content"
            id="mdc-dialog-content"
          >
            <div
              className="mdc-text-field"
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "marginBottom": "20px",
                  "width": "100%",
                }
              }
            >
              <input
                className="mdc-text-field__input"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                placeholder={null}
                value=""
              />
              <label
                className="mdc-floating-label"
                htmlFor=""
                onAnimationEnd={[Function]}
              >
                Factory Name
              </label>
              <div
                className="mdc-line-ripple"
                onTransitionEnd={[Function]}
                style={Object {}}
              />
            </div>
            <div
              className="mdc-text-field"
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "marginBottom": "20px",
                  "width": "100%",
                }
              }
            >
              <input
                className="mdc-text-field__input"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                placeholder={null}
                value={1}
              />
              <label
                className="mdc-floating-label mdc-floating-label--float-above"
                htmlFor=""
                onAnimationEnd={[Function]}
              >
                Min Value
              </label>
              <div
                className="mdc-line-ripple"
                onTransitionEnd={[Function]}
                style={Object {}}
              />
            </div>
            <div
              className="mdc-text-field"
              onClick={[Function]}
              onKeyDown={[Function]}
              style={
                Object {
                  "marginBottom": "inherit",
                  "width": "100%",
                }
              }
            >
              <input
                className="mdc-text-field__input"
                disabled={false}
                id=""
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                placeholder={null}
                value={500}
              />
              <label
                className="mdc-floating-label mdc-floating-label--float-above"
                htmlFor=""
                onAnimationEnd={[Function]}
              >
                Max Value
              </label>
              <div
                className="mdc-line-ripple"
                onTransitionEnd={[Function]}
                style={Object {}}
              />
            </div>
            <div
              className="mdc-chip"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              onTransitionEnd={[Function]}
              style={
                Object {
                  "border": "1px solid red",
                  "marginTop": "20px",
                }
              }
              tabIndex={0}
            >
              <div
                className="mdc-chip__text"
              >
                Name range is 1 to 128 chars
              </div>
            </div>
          </div>
          <footer
            className="mdc-dialog__actions"
          >
            <button
              className="mdc-button mdc-dialog__button"
              data-mdc-dialog-action="save"
              disabled={true}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
            >
              <span
                className="mdc-button__label"
              >
                Save
              </span>
            </button>
            <button
              className="mdc-button mdc-dialog__button"
              data-mdc-dialog-action="dismiss"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
            >
              <span
                className="mdc-button__label"
              >
                Cancel
              </span>
            </button>
          </footer>
        </div>
      </div>
      <div
        className="mdc-dialog__scrim"
      />
    </div>
  </div>,
]
`;

exports[`Add Factory Button it renders as button 1`] = `
<button
  className="mdc-fab"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "bottom": 20,
      "position": "fixed",
      "right": 20,
      "zIndex": 1,
    }
  }
>
  <i
    className="material-icons mdc-fab__icon"
  >
    add
  </i>
</button>
`;
